src/sentry/seer/endpoints/organization_seer_onboarding.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import annotations
+
+import logging
+
+from rest_framework import serializers
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from sentry.api.api_owners import ApiOwner
+from sentry.api.api_publish_status import ApiPublishStatus
+from sentry.api.base import region_silo_endpoint
+from sentry.api.bases import OrganizationEndpoint, OrganizationPermission
+from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+from sentry.models.organization import Organization
+from sentry.seer.autofix.autofix import onboarding_seer_settings_update
+from sentry.seer.endpoints.project_seer_preferences import BranchOverrideSerializer
+from sentry.seer.models import SeerRepoDefinition
+
+logger = logging.getLogger(__name__)
+
+
+class RepositorySerializer(CamelSnakeSerializer):
+    provider = serializers.CharField(required=True)
+    owner = serializers.CharField(required=True)
+    name = serializers.CharField(required=True)
+    external_id = serializers.CharField(required=True)
+    organization_id = serializers.IntegerField(required=False, allow_null=True)
+    integration_id = serializers.CharField(required=False, allow_null=True)
+    branch_name = serializers.CharField(required=False, allow_null=True)
+    branch_overrides = BranchOverrideSerializer(many=True, required=False, default=list)
+    instructions = serializers.CharField(required=False, allow_null=True, allow_blank=True)
+    base_commit_sha = serializers.CharField(required=False, allow_null=True)
+    provider_raw = serializers.CharField(required=False, allow_null=True)
```

Comment:
maps to https://github.com/getsentry/sentry/blob/bbe0c776650931db01e747ee6a70ea9811beee09/src/sentry/seer/models.py#L29-L42